biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java

```diff
@@ -366,8 +367,18 @@ private Processor loadRepositories() throws Exception {
 			// Map the repository names...
 
 			Map<String,Repository> repoNameMap = new HashMap<String,Repository>(allRepos.size());
-			for (Repository repo : allRepos)
-				repoNameMap.put(repo.toString(), repo);
+			for (Repository repo : allRepos) {
+				String name = repo.toString();
+				if (repo instanceof RepositoryPlugin) {
+					name = ((RepositoryPlugin) repo).getName();
+				} else if (repo instanceof aQute.bnd.deployer.repository.wrapper.Plugin) {
```

Comment:
You may want to order the tests such that the wrapper plugin is tested for first since it is the most specific type. Right now it is not a RepositoryPlugin, but some super type could be changed to implement RepositoryPlugin and then it would fall into the if RepositoryPlugin test and miss the, I assume necessary, call to init before toString.